### DISCUSSION TITLE
Webview2 new window

### DISCUSSION THREAD

**Discussion by mohanbabu17** [MicrosoftEdge/WebView2Feedback#2481] [Q&A]
Hello Friends,

I am using webview2 in my windows form, i have  popup window which I captured in main window by below code

1) 
Private Sub CoreWebView2_NewWindowRequested(ByVal sender As Object, ByVal e As Microsoft.Web.WebView2.Core.CoreWebView2NewWindowRequestedEventArgs)
        e.Handled = True
        Dim popup = New FrmPopUp
        popup.Deferral = e.GetDeferral()
        popup.EventArgs = e
        popup.WebView21.Source = New Uri(e.Uri)
        popup.Show()
        popup.Activate()
    End Sub

2) 
And in popup window i have below Initialization code:

Private Sub WebView21_CoreWebView2InitializationCompleted(sender As Object, e As CoreWebView2InitializationCompletedEventArgs) Handles WebView21.CoreWebView2InitializationCompleted
        If e.IsSuccess Then
            ' Was Deferral set in NewWindowRequested?
            If Deferral IsNot Nothing Then
                ' Set this instance, complete NewWindowRequested And clear deferral
                EventArgs.NewWindow = WebView21.CoreWebView2

                Deferral.Complete()
                Deferral = Nothing
                EventArgs = Nothing
            End If
        End If
    End Sub

3)
 Issue i face in popup NavigationCompleted, this event is triggered only first time and next event is not triggering.

Please help me.

**Comment by victorthoang** [comment]
hello @mohanbabu17 

Are you still looking for an answer to your question? If you still need a response, we can turn this into an Issue with a Question tag (which is typically how we address questions and bugs.

The discussion thread might be more useful for off topic questions or non-issue related inquiries!

Let me know if you'd like me to create an issue from this discussion, or if you'd prefer to add it. Thanks!
